flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer08.java

```diff
@@ -125,4 +125,17 @@ public FlinkKafkaProducer08(String topicId, KeyedSerializationSchema<IN> seriali
 		super(topicId, serializationSchema, producerConfig, customPartitioner);
 	}
 
+	@Override
+	protected void flush() {
+		// The Kafka 0.8 producer doesn't support flushing, therefore, we are using an inefficient
+		// busy wait approach
+		while(pendingRecords > 0) {
+			try {
+				Thread.sleep(10);
```

Comment:
I'm not a big fan of busy waiting. Can't we wait on the lock for accessing `pendingRecords` and let the callbacks call `notify` when the `pendingRecords == 0`?